torch_xla/csrc/init_python_bindings.cpp

```diff
@@ -742,6 +744,155 @@ void BuildProfilerSubmodule(py::module* m) {
       });
 }
 
+class PyLoweringContext {
+ public:
+  PyLoweringContext() : PyLoweringContext(GetCurrentDevice()) {}
+
+  PyLoweringContext(torch::lazy::BackendDevice device)
+      : lowering_ctx("PyLoweringContext", device) {}
+
+  // Builds a HLO graph given a set of output tensors. It is assumed that the
+  // tensors passed to this function are the result of the forward pass of a
+  // given Module/function.
+  void Build(std::vector<at::Tensor> tensors) {
+    // Get the backing XLA tensors from the output torch tensor handles
+    std::vector<XLATensorPtr> xtensors = GetXlaTensors(tensors, /*want_all=*/true);
+
+    // Get the lazy IR value from the output XLA tensors
+    std::vector<torch::lazy::Value> ir_values;
+    for (auto& xtensor : xtensors) {
+      torch::lazy::Value value = xtensor->CurrentIrValue();
+      ir_values.push_back(value);
+    }
```

Comment:
what if `CurrentIrValue` return a defualt value? this call here assumes all xtensor will have a IR value which is not necessary true.

Reply:
Under what conditions does an xtensor not have an IR value? 

I am aware that this could cause problems if you had both CPU/XLA allocated tensors, but is there another condition where an IR value would not exist?

I think the correct behavior would be that if you were passed a tensor that did not have an IR value then we should exit with an exception (unless this is otherwise recoverable?)

Reply:
XLATensor can be in many format, it can has a `XLAData` which means it is a data on device, it can be a `IR Value` which means it is an computation to be computed. It can also be `at::Tensor` means it is still a tensor on host and we haven't upload the tensor to device yet.

If you want to force the IR value(XLAData and `at::tensor` can become IR, which is a DeviceData IR), you want to use `GetIrValue` instead of `CurrentIRValue`